File: new/74_search_a_2d_matrix_brute_force.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        // idea: brute force
        for (int row = 0; row < matrix.size(); row++) {
            for (int column = 0; column < matrix[row].size(); column++) {
                if (matrix[row][column] == target) {
                    return true;
                }
            }
        }

        return false;
    }
    // time: O(n * m) - n = number of rows, m = number of columns
    // space: O(1)
};

File: new/python/74_search_a_2d_matrix_brute_force.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        # idea: brute force
        for row in range(len(matrix)):
            for column in range(len(matrix[row])):
                if matrix[row][column] == target:
                    return True
        return False
        # time: O(m * n)
        # space: O(1)
